src/components/panes/JoinSubmissionPane.jsx

```diff
@@ -151,7 +152,15 @@ export default class JoinSubmissionPane extends PaneBase {
                             label = this.props.intl.formatMessage({ id: `misc.fields.${fieldName}` });
                         }
 
-                        const value = person[fieldName] || '';
+                        let value;
+                        if(fieldName == 'gender') {
+                            value = this.props.intl.formatMessage({ id: `misc.fields.genderOptions.${person[fieldName]}` })
```

Comment:
If we are worried that the value is not one of the four permitted values (`'f'`, `'m'`, `'o'`, `null`) we should probably fall back anything that isn't f/m/o to some other value and use that as the message ID (e.g. `'misc.fields.genderOptions.unknown'`).